java/flight/src/test/java/org/apache/arrow/flight/TestClientMiddleware.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.flight;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.BiConsumer;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * A basic test of client middleware using a simplified OpenTracing-like example.
+ */
+@RunWith(JUnit4.class)
+public class TestClientMiddleware {
+
+  @Test
+  public void rejectCall() {
+    test(new Producer(), null, Collections.singletonList(new CallRejector.Factory()),
+        (allocator, client) -> {
+          FlightTestUtil.assertCode(FlightStatusCode.UNAVAILABLE, client::listActions);
+        });
+  }
+
+  @Test
+  public void injectSpans() {
```

Comment:
this test could use some more documentation, it is a little sense to read as is.

Reply:
I've renamed/cleaned up the tests here. The main scenario being tested is an OpenTracing-like example where a request ID gets invisibly propagated when you make a request.